workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/views/matplotlib.py

```diff
@@ -0,0 +1,84 @@
+import base64
+import dataclasses
+import io
+import logging
+from collections import defaultdict
+from typing import Iterable, Sequence
+
+import jinja2
+import matplotlib.pyplot as plt
+import xarray as xr
+
+
+def fig_to_b64(fig, format="png"):
+    pic_IObytes = io.BytesIO()
+    fig.savefig(pic_IObytes, format=format)
+    pic_IObytes.seek(0)
+    pic_hash = base64.b64encode(pic_IObytes.read())
+    return f"data:image/png;base64, " + pic_hash.decode()
+
+
+@dataclasses.dataclass
+class raw_html:
+    contents: str
+
+    def __repr__(self):
+        return self.contents
+
+
+template = jinja2.Template(
+    """
+<h2> {{varfilter}} </h2>
+<table>
+
+<tr>
+{% for run in runs %}
+<td><center> {{ run }} </center></td>
+{% endfor %}
+</tr>
+
+{% for varname, runs in diagnostics.items() %}
+{% for run, src in runs.items() %}
+<td>
+<img src="{{src}}" width="500px" />
+</td>
+{% endfor %}
+</tr>
+{% endfor %}
+</table>
```

Comment:
This can lead to figures which end up in the wrong column if there is a run which is missing a variable. Maybe loop over `runs` as for the headers, and just insert some empty space as a placeholder if the variable is not available?

As an example, see the lowest layer temperature row [here](https://00f74ba44be0b90c09dad45bbf881153e23a6160c3-apidata.googleusercontent.com/download/storage/v1/b/vcm-ml-scratch/o/test-prognostic-report%2F47de08ea676e%2Fhovmoller.html?jk=AFshE3WBRG9b_CDbVMYmi2_LMWWlGDrL6CvvhwRIgrK2lDlqYSFzU0LuIC6ekB0u5kdKcFW2Ufbz7ugbk0bBmpnaDvd_UcPzTepKwraivmjhZyKjW8lAiBFM2OVU6pSJQckfCbwegafrwk05RhyWS6DavPOYat_tq_r1igvV9vjlsNvPPayfS0lyU_0luCv0aEr-bbTKXnC79Gg4AexKHYuc1d9tgW_EKvvLokjqtI8uKN8JVkXBSnAXtnRfYylkAkskpMlDdIRCFVUIO1H6VZ-WukVQriMh8f7Gx64DEQBzkTr9Pm49GMOP5jkyF8SV2324jyqsIzp_Y7o8yhoH7kJgX7xHVbEQGrfaBb__cKfvhYXEnAGaWhD0tP2Dpm7EmjgEcsPffn-1YsMz3Oc7NF3moJ8ECHjCNIIH5Z3eU35_vu2IJvwS-Ak-jVo4PjAqocTzAb26aGWUnknL2rgD8IcXwkUUKRVmHsS5MMxJldQ7yWv_60kZ2ZVGfQ6sqdBgcFRWvN7CguV1r7RhbFJqAETIoBQsAgWZp1zi25yFAEcCnwlfoYeqhcFqg_M7fCGFtNqGQfOqkNQAaiesDAqMzh9BogFAlF0I_CWq2quJV5rRJTHbJODoSmTcv82Fea6JnMOmaUr7Tw56oDxB0LEpXMEz_CBvvvm2gV6SnQvs-vYqE2rpJDLReJF_926h-mYbK-t7p0x6wN_aOoYOel3tcCDDJORcOMzLJWzxuz8Q3o3ospwVNMGboK2XUSc8kzCzH7b28X8TyeZcQSxV5_XHob1gFWqxMXWgGX649TqyFPGM66zHysu2dVeVIgv3SLoJTvK03lV46YrSszb6djPqQI_hmMTqJy7Y1onMb1YUBrl_ik6XLmHItDRWkJfR2OX46gDH5IA1HScqqS-VVjeEEIg1tPd3OEnRhuPxuZi0niImQNiNMd3CNDXy69r2R8MAhy7OYl-zAIVGErPA9avFez1zQdrqJpgPCzuhTI_JjRFDzwx6JVg1UcO0qucnF00&isca=1) (the "verification" figure should be missing, not the "run2" figure).

Reply:
I fixed this using the `RunDiagnostics.get_variable`.